### PR TITLE
curl: 7.52.1 -> 7.53.0

### PR DESCRIPTION
**Pull request by NeQuissimus**
###### Motivation for this change
Update

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/curl/default.nix [hunk 1]
```diff
@@ -21,14 +21,14 @@ assert scpSupport -> libssh2 != null;
...
-  name = "curl-7.52.1";
+  name = "curl-7.53.0";
 
   src = fetchurl {
     url = "http://curl.haxx.se/download/${name}.tar.bz2";
-    sha256 = "16rqhyzlpnivifin8n7l2fr9ihay9v2nw2drsniinb6bcykqaqfi";
+    sha256 = "008833dd9w4l2277q9r0bsq1vqmm0fr7qqyzvqlw5d47xy5mld5j";
   };
 
-  patches = [ ./issue-1174.patch ];
```
Comment (ndowens) [inline pkgs/tools/networking/curl/default.nix]: Why not remove patches = [ ]; when empty?

Comment (NeQuissimus) [inline pkgs/tools/networking/curl/default.nix]: we tend to pop patches in and out as CVEs come up, just makes it easier...

Comment (ndowens) [inline pkgs/tools/networking/curl/default.nix]: Ah did not know that :)

**Comment by NeQuissimus** [comment]
lol, @globin gave me a heart-attack when all these commits showed up here... I don't think we staged curl last time, did we?

**Comment by globin** [comment]
I think the last time was a security update, generally as I am quite sure this is a mass-rebuild it should go through staging.

**Comment by edolstra** [comment]
This PR is showing 64 mostly unrelated commits, looks like something went wrong here.

**Comment by vcunat** [comment]
It shows additional stuff from `staging..master`.

**Comment by globin** [comment]
This was originally based on master and the commits that are on master, but not on staging are showing up here, too.

**Comment by vcunat** [comment]
I believe this actually **is** a security fix: https://curl.haxx.se/docs/adv_20170222.html

**Pull request merged**